sycl/plugins/level_zero/tracing.cpp

```diff
@@ -0,0 +1,145 @@
+      xptiNotifySubscribers(                                                   \
+          CallStreamID, (uint16_t)xpti::trace_point_type_t::function_begin,    \
+          GCallEvent, nullptr, CallCorrelationID, FuncName);                   \
+      uint32_t FuncID = static_cast<uint32_t>(ZEApiKind::call);                \
+      xpti::function_with_args_t Payload{FuncID, FuncName, Params, nullptr,    \
+                                         nullptr};                             \
+      xptiNotifySubscribers(                                                   \
+          DebugStreamID,                                                       \
+          (uint16_t)xpti::trace_point_type_t::function_with_args_begin,        \
+          GDebugEvent, nullptr, DebugCorrelationID, &Payload);                 \
+    }                                                                          \
+  };                                                                           \
+  Epilogue.domain.cb = [](params_type *Params, ze_result_t Result, void *,     \
+                          void **) {                                           \
+    if (xptiTraceEnabled()) {                                                  \
+      uint8_t CallStreamID = xptiRegisterStream(L0_CALL_STREAM_NAME);          \
+      uint8_t DebugStreamID = xptiRegisterStream(L0_DEBUG_STREAM_NAME);        \
+      const char *FuncName = #call;                                            \
+      xptiNotifySubscribers(CallStreamID,                                      \
+                            (uint16_t)xpti::trace_point_type_t::function_end,  \
+                            GCallEvent, nullptr, CallCorrelationID, FuncName); \
+      uint32_t FuncID = static_cast<uint32_t>(ZEApiKind::call);                \
+      xpti::function_with_args_t Payload{FuncID, FuncName, Params, &Result,    \
+                                         nullptr};                             \
+      xptiNotifySubscribers(                                                   \
+          DebugStreamID,                                                       \
+          (uint16_t)xpti::trace_point_type_t::function_with_args_end,          \
+          GDebugEvent, nullptr, DebugCorrelationID, &Payload);                 \
+    }                                                                          \
+  };
+
+#include "ze_api.def"
+
+#undef _ZE_API
+
+  Status = zelTracerSetPrologues(Tracer, &Prologue);
+  if (Status != ZE_RESULT_SUCCESS) {
+    std::cerr << "Failed to enable L0 tracing\n";
+    std::terminate();
+  }
+  Status = zelTracerSetEpilogues(Tracer, &Epilogue);
+  if (Status != ZE_RESULT_SUCCESS) {
+    std::cerr << "Failed to enable L0 tracing\n";
+    std::terminate();
+  }
+
+  Status = zelTracerSetEnabled(Tracer, true);
+  if (Status != ZE_RESULT_SUCCESS) {
+    std::cerr << "Failed to enable L0 tracing\n";
+    std::terminate();
+  }
+#endif
+}
+
+void disableL0Tracing() {
```

Comment:
Should we call this function in piTearDown?

Reply:
Yes, I think we should. Added.

Reply:
Sergey, could you please push this update.